### DISCUSSION TITLE
Missing authorization header

### DISCUSSION THREAD

**Discussion by bondjw07** [etsy/open-api#914] [Q&A] [answered]
I feel like I am losing my mind.  Whenever I make a call to a resource that includes a `shop_id` as part of the route, I am getting an 401 with the message `Missing authorization header`.

I am including my KEYSTRING from my application in a header of type `x-api-key`, and I am able to successfully call other routes such as `application/seller-taxonomy/nodes`.  

This is not a commercial app if that makes any difference.

**Comment by PMStanley** [comment, marked as answer]
You'll need to provide an example of a specific endpoint you're calling and the code you're using to call it but if I had to hazard a guess you're not sending the Authorization header with your access token:

> const requestOptions = {
        headers: {
            'x-api-key': clientID,
            // Scoped endpoints require a bearer token
            Authorization: `Bearer ${access_token}`,
        }
    };

>

**Reply by bondjw07** [reply]
Thanks @PMStanley. I'm indeed not passing an OAuth2 Bearer access token.  I have not had any luck authenticating with a non-commercial app, do you know if that authentication method is limited to commercial apps? 

Edit: for more detail, I'm getting:`Error: Invalid authorization header` when making a call to `https://openapi.etsy.com/v3/public/oauth/token`
<img width="841" alt="image" src="https://user-images.githubusercontent.com/5480627/225785874-626186a9-f40c-4c25-b52c-54e88d96db66.png">

**Reply by bondjw07** [reply]
Here's why I think it might not be 
<img width="557" alt="image" src="https://user-images.githubusercontent.com/5480627/225783575-440b341c-27c3-47fa-a03f-a5a9875020b1.png">

**Reply by PMStanley** [reply]
@bondjw07 You need to submit the request for OAuth access, if I remember correctly you will need to explain why you need access but won't need to check the 'Commercial access' box.

**Reply by bondjw07** [reply]
Thanks @PMStanley.  Any idea how I might go about requesting that access after the app is in "Personal Access" status?  The only request button is for Commercial Access.  Wondering if I need to directly get in touch with support somehow.

**Reply by PMStanley** [reply]
When I went through the process I was able to request access without enabling full access to the public, I'm not clear if the process has changed but I'd have a look what options are available when you press that button.  

That said, the docs suggest that you should have OAuth token access with the 'personal access' tier so dropping a note to support wouldn't be a bad idea.
